python/pyspark/serializers.py

```diff
@@ -523,7 +523,21 @@ def namedtuple(*args, **kwargs):
         for k, v in _old_namedtuple_kwdefaults.items():
             kwargs[k] = kwargs.get(k, v)
         cls = _old_namedtuple(*args, **kwargs)
-        return _hack_namedtuple(cls)
+
+        import sys
+        f = sys._getframe(1)
```

Comment:
Hm .. https://docs.python.org/2/library/sys.html#sys._getframe

> CPython implementation detail: This function should be used for internal and specialized purposes only. It is not guaranteed to exist in all implementations of Python.

Is it safe to use it?

Reply:
Good point. [`collections.nametuple`](https://github.com/python/cpython/blob/master/Lib/collections/__init__.py#L466) has a fix for Jython and IronPython. I can backport it for completeness, but realistically, the probability of someone running PySpark on these implementations is not very high.

Reply:
Yea but thing is, that the doc says this not guaranteed although most of Python implementations look having it - there's a risk here we should take (it could be broken in a specific implementation of Python although it sounds unlikely). Is there any other way to avoid this?

Reply:
Another way to fix the module name is to get rid of the extra stack frame. This can be done by e.g. modifying the bytecode of `collections.namedtuple` so that it redirected to `_hack_namedtuple` when needed. However, I think that the current implementation (despite being magical) is better than bytecode hacking since it is as magical as `collections.namedtuple` itself.

I can change the PR to do the same as `collections.namedtuple` for cross-interpreter compatibility, wdyt?